src/Umbraco.Web/Editors/TinyMceController.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Web.Http;
+using Umbraco.Core.Services;
+using Umbraco.Web.WebApi;
+using Umbraco.Core;
+using Umbraco.Web.Mvc;
+using Umbraco.Core.IO;
+using System.IO;
+using System.Threading.Tasks;
+using Umbraco.Web.Composing;
+using Umbraco.Core.Configuration.UmbracoSettings;
+using System.Linq;
+using System;
+
+namespace Umbraco.Web.Editors
+{
+    [PluginController("UmbracoApi")]
+    public class TinyMceController : UmbracoAuthorizedApiController
+    {
+        private IMediaService _mediaService;
+        private IContentTypeBaseServiceProvider _contentTypeBaseServiceProvider;
+
+        public TinyMceController(IMediaService mediaService, IContentTypeBaseServiceProvider contentTypeBaseServiceProvider)
+        {
+            _mediaService = mediaService;
+            _contentTypeBaseServiceProvider = contentTypeBaseServiceProvider;
+        }
+
+        [HttpPost]
+        public async Task<HttpResponseMessage> UploadImage()
+        {
+
+            if (Request.Content.IsMimeMultipartContent() == false)
+            {
+                throw new HttpResponseException(HttpStatusCode.UnsupportedMediaType);
+            }
+
+            // Backoffice user ID (needed for unique folder path) to help with concurrent users
+            // to avoid filename clash along with UTC current time
+            var userId = Security.CurrentUser.Id;
+            var imageTempPath = IOHelper.MapPath(SystemDirectories.TempImageUploads + "/" + userId + "_" + DateTimeOffset.UtcNow.ToUnixTimeSeconds());
```

Comment:
Why not just use a guid ? Can't see you need to user ID, and en theory in a load balanced setup the time could be a little off. I think a guid is simpler and safer.